src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -5999,6 +6103,33 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 tmp = make_call_helper(al, fn_div, current_scope, args, "_lpython_str_endswith", loc);
             }
             return;
+        } else if (attr_name == "partition") {
+            /*  
+                str.partition(seperator)        ---->
+                Split the string at the first occurrence of sep, and return a 3-tuple containing the part 
+                before the separator, the separator itself, and the part after the separator. 
+                If the separator is not found, return a 3-tuple containing the string itself, followed 
+                by two empty strings.
+            */
+            if (args.size() != 1) {
+                throw SemanticError("str.partition() takes one arguments",
+                        loc);
+            }
+            ASR::expr_t *arg_seperator = args[0].m_value;
+            ASR::ttype_t *arg_seperator_type = ASRUtils::expr_type(arg_seperator);
+            if (!ASRUtils::is_character(*arg_seperator_type)) {
+                throw SemanticError("str.partition() takes one arguments of type: str",
+                    arg_seperator->base.loc);
+            }
+            if(s_var.size() == 0) {
+                throw SemanticError("string to undergo partition cannot be empty",loc);
+            }
+            ASR::expr_t *value = nullptr;
+            if(ASRUtils::expr_value(arg_seperator)) {
+                value = evaluate_partition(s_var, arg_seperator, loc, arg_seperator_type);
+            }
+            create_partition(loc, s_var, arg_seperator, arg_seperator_type, value);
```

Comment:
The changes look good to me!

But, I think the check with the function call (`evaluate_partition`) can be moved into the `create_partition` function.
We do the same in the IntrinsicFunction `eval_*` and `create_*`

Reply:
Okay. I will move it inside `create_partition`. I did this because certik above suggested to do this way.

Reply:
Done with the changes. @Thirumalai-Shaktivel